### ISSUE TITLE
Support authentication for plugin repositories 

### ISSUE THREAD

**Issue by jb-2020** [gradle/actions#31]
In https://github.com/gradle/gradle-build-action/issues/933 the gradle configurations were enhanced to support a custom plugin repository provided by the user.  These plugin repositories can require authentication but there is currently no mechanism to also provide a credentials block to this configuration. 

It would be helpful if we could also provide a username and password to the custom plugin repository e.g.
```
  repositories {
    maven { 
      url "<private gradle plugin repository>
      credentials {
        username = System.getenv("USERNAME")
        password = System.getenv("PASSWORD")
      }
    }
  }
```

**Comment by bigdaz** [comment]
Fixed in #61

**Comment by bigdaz** [comment]
Also fixed in 7b589d97403cedc7180e3ebc56a1c1a860fa54f5
